### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by seppo0010**

### CODE REVIEW

File: web/ts/create-index.ts [hunk 1]
```diff
@@ -24,6 +25,6 @@ const run = async () => {
     }
   }
 
-  await fs.writeFile(path.join(DATADIR, 'ms-index.json'), JSON.stringify(ms.toJSON()))
+  await fs.writeFile(process.argv[3], JSON.stringify(ms.toJSON()))
 }
 run()
```
Comment (xaiki) [inline web/ts/create-index.ts:30]: if you're going to do this, you need to add a default yarn run that would do something usefull.

File: run.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+./builder -blacklist=/tmp/juscaba/blacklist "-json=/tmp/juscaba/${exp_filename}.json" -pdfs=/tmp/juscaba/pdfs "-expediente=${exp}" -images=true
+
+pushd ts
+yarn run ts-node create-index.ts /tmp/juscaba/${exp_filename}.json /tmp/juscaba/${exp_filename}-index.json
+popd
+
+cp /tmp/juscaba/${exp_filename}{-index,}.json ${SCRIPT_DIR}/web/public/data
+
+yarn build
+mv build/* /tmp/juscaba/web
```
Comment (xaiki) [inline run.sh:20]: you could prob call this from ts/ npm/yarn run

File: builder/main.go [hunk 3]
```diff
@@ -12,6 +13,46 @@ import (
...
+		return nil, err
+	}
+
+	args.blacklist, err = readBlacklist(blacklistPath)
+	if err != nil {
+		return nil, err
+	}
+	return &args, nil
+}
+
```
Comment (xaiki) [inline builder/main.go:55]: should this go to another commit ?

File: builder/extracttext/lib.go [hunk 4]
```diff
@@ -198,10 +198,15 @@ func GetDocumentText(r io.Reader) (string, error) {
...
-		return "", err
+	text2 := ""
+	if images {
+		log.Info("getting pdf images text")
+		text2, err = getDocumentImagesText(dir, p)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		log.Info("skipping images text")
```
Comment (xaiki) [inline builder/extracttext/lib.go:209]: other commit ?

**Review by xaiki — CHANGES_REQUESTED** [review]

File: run.sh (hunk 2 above)
Comment (seppo0010) [reply run.sh:20]: Could I? npm/yarn creates a `build` directory inside the project. This moves it to a directory that's only going to exist within the dockerfile so it can be mounted. If I move this to a yarn script, the project would not run outside docker. I think.

File: Dockerfile [hunk 5]
```diff
@@ -1,29 +1,35 @@
...
 
-COPY ./builder/go.mod ./builder/go.sum ./
-RUN go mod download
-
-COPY ./builder .
-
-RUN go get -d -v ./...
-RUN go install -v ./...
+WORKDIR /app/ts
+RUN yarn
```
Comment (xaiki) [inline Dockerfile:29]: nitpick: here is where build-npm stops

**Review by xaiki — APPROVED** [review]
lgtm

File: Dockerfile (hunk 5 above)
Comment (seppo0010) [reply Dockerfile:29]: what do you mean?

Comment (xaiki) [reply Dockerfile:29]: you could have split that in another container
- build-go
- build-npm
- run
